internal/build/phases.go

```diff
@@ -29,6 +29,66 @@ type PhaseFactory interface {
 	New(provider *PhaseConfigProvider) RunnerCleaner
 }
 
+func (l *Lifecycle) Create(ctx context.Context, publish, clearCache bool, runImage, launchCacheName, cacheName, repoName, networkMode string, phaseFactory PhaseFactory) error {
+	var configProvider *PhaseConfigProvider
+	var binds []string
+
+	args := []string{
+		"-run-image", runImage,
+		repoName,
+	}
+
+	if clearCache {
+		args = append([]string{"-skip-restore"}, args...)
+	} else {
+		args = append([]string{"-cache-dir", cacheDir}, args...)
```

Comment:
As far as I understand, `restore` is being called only when the `clearCache` flag is set to true (as can be seen [here](https://github.com/buildpacks/pack/blob/dd793414d1094f42b2b0671d0a9171091614f13c/internal/build/build.go#L107-L112)).
I think it makes sense since the restore phase is responsible on restoring any previously-cached dependencies but we can skip it.
I think it's similar with the export phase but I'm not sure and I don't see a proof in the code.
Anyway I thought that the `creator` may or may not use the cache.
Maybe @ekcasey or @jromero can clarify that.

Reply:
Ah, that makes sense about restore. I think we'll want to confirm about export though.

Reply:
So you were right!
We need to pass the `-cache-dir` even when `clearCache` is set to true.
Otherwise, the `cacheStore` will be `nil` [here](https://github.com/buildpacks/lifecycle/blob/3dde35fd5d4f1a3a3412eac7b819fa597d597cd9/cmd/lifecycle/exporter.go#L215) and therefore we won't get to "Adding cache layer" [here](https://github.com/buildpacks/lifecycle/blob/3dde35fd5d4f1a3a3412eac7b819fa597d597cd9/exporter.go#L321).
Please see [this commit](https://github.com/buildpacks/pack/pull/595/commits/b6961bf7c99560fcc5dd0a9deb1ed607dd876aa9) that fixes this issue.
Thanks!